2020-11-10  Dhakshin Suriakannu  <dhakshin@lnxsrv06.seas.ucla.edu>

        * timespec.h (timespec_cmp): Added an if statement to compare the time of file
	a and b to return either 1, -1, or 0 based on which time is greater

--- timespec2.h	2020-11-10 21:29:46.321528000 -0800
+++ timespec.h	2020-11-10 21:30:10.146675000 -0800
@@ -45,8 +45,19 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+
+if (a.tv_sec > b.tv_sec)
+  {
+    return 1;
+  }
+  else if (a.tv_sec < b.tv_sec)
+  {
+    return -1;
+  }
+  else
+  {
+    return 0;
+  }
 }
 
 # if ! HAVE_DECL_NANOSLEEP
